src/topology/omega_complete_partial_order.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2020 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+-/
+import topology.basic
+import order.omega_complete_partial_order
+
+/-!
+# Scott Topological Spaces
+
+I don't know what they're good for but their notion
+of continuity is equivalent to continuity in ωCPOs.
+
+## Reference
+
+ * https://ncatlab.org/nlab/show/Scott+topology
+
+-/
+
+open omega_complete_partial_order
+open_locale classical
+
+universes u
+namespace scott_topological_space
+
+/--  -/
+def is_ωSup {α : Type u} [preorder α] (c : chain α) (x : α) : Prop :=
+(∀ i, c i ≤ x) ∧ (∀ y, (∀ i, c i ≤ y) → x ≤ y)
+
+variables (α : Type u) [omega_complete_partial_order α]
+
+/-- The characteristic function of open sets is monotone and preserves
+the limits of chains. -/
+def is_open (s : set α) : Prop :=
+continuous' (s : α → Prop)
+
+theorem is_open_univ : is_open α set.univ :=
+⟨λ x y h, le_refl _,
+  show omega_complete_partial_order.continuous ⊤,
+    from complete_lattice.top_continuous ⟩
+
+theorem is_open_inter (s t : set α) : is_open α s → is_open α t → is_open α (s ∩ t) :=
+begin
+  simp only [is_open, exists_imp_distrib, continuous'],
+  intros h₀ h₁ h₂ h₃,
+  rw ← set.inf_eq_inter,
+  let s' : α →ₘ Prop := ⟨s, h₀⟩,
+  let t' : α →ₘ Prop := ⟨t, h₂⟩,
+  split, change omega_complete_partial_order.continuous (s' ⊓ t'),
+  haveI : is_total Prop (≤) := ⟨ @le_total Prop _ ⟩,
+  apply complete_lattice.inf_continuous; assumption
+end
+
+theorem is_open_sUnion : ∀s, (∀t∈s, is_open α t) → is_open α (⋃₀ s) :=
+begin
+  introv h₀,
+  suffices : is_open α (Sup s : α → Prop),
+  { convert this, ext,
+    simp only [set.sUnion, Sup, set_of, supr, conditionally_complete_lattice.Sup,
+               complete_lattice.Sup, exists_prop, set.mem_range,
+               set_coe.exists, eq_iff_iff, subtype.coe_mk],
+    tauto, },
+  apply complete_lattice.Sup_continuous' _ h₀,
+end
+
+end scott_topological_space
+
+/-- A Scott topological space is defined on preorders
+such that their open sets, seen as a function `α → Prop`,
+preserves the joins of ω-chains  -/
+@[reducible]
+def Scott (α : Type u) := α
+
+instance scott_topological_space (α : Type u) [omega_complete_partial_order α] : topological_space (Scott α) :=
+{ is_open := scott_topological_space.is_open α,
+  is_open_univ := scott_topological_space.is_open_univ α,
+  is_open_inter := scott_topological_space.is_open_inter α,
+  is_open_sUnion := scott_topological_space.is_open_sUnion α }
+
+@[simp]
+lemma set_of_apply {α} (p : α → Prop) (x : α) : set_of p x ↔ p x := iff.rfl
```

Comment:
Should this be in this file?

Reply:
This should not exist at all, it is evidence of abstraction breaking.